File: practice/data_structures/trees/preorder_traversal/preorder_traversal.py
"""
Node is defined as
self.left (the left child of the node)
self.right (the right child of the node)
self.data (the value of the node)
"""

def preOrder(i_root):
    if i_root:
        print i_root.data,
        preOrder(i_root.left)
        preOrder(i_root.right)

File: practice/data_structures/trees/preorder_traversal/preorder_traversal.cpp
/* you only have to complete the function given below.  
Node is defined as  

struct node
{
    int data;
    node* left;
    node* right;
};

*/

void preOrder(node * i_root)
{
    if (i_root)
    {
        cout << i_root->data << " ";
        preOrder(i_root->left);
        preOrder(i_root->right);        
    }
}

